libraries/AdaptiveExpressions/BuiltinFunctions/Rand.cs

```diff
@@ -43,9 +42,28 @@ private static EvaluateExpressionDelegate Evaluator()
                             }
                             else
                             {
-                                lock (_randomizerLock)
+                                if (options.Properties.TryGetValue("randomValue", out var randomValue)
+                                && randomValue.IsInteger())
+                                {
+                                    var randomValueNum = Convert.ToInt32(randomValue, CultureInfo.InvariantCulture);
+                                    value = min + (randomValueNum % (max - min));
+                                }
+                                else
                                 {
-                                    value = Randomizer.Next(min, max);
+                                    var random = new Random();
+                                    if (options.Properties.TryGetValue("randomSeed", out var randomSeed))
```

Comment:
Creating a random is relatively expensive and the purpose of a seed is to start a whole sequence of random values.  You also would not need a lock if you create it on the fly.  What you should do: 
1) Have a single static random.
2) If there is no seed or it is a new seed, create a new one.
3) When you use it, do a lock.